src/device/mod.rs

```diff
@@ -1492,7 +1492,7 @@ impl Runtime {
             .meshnet_config
             .as_ref()
             .and_then(|config| config.peers.clone())
-            .and_then(|config_peers| {
```

Comment:
This whole logic can be simplified for sure, possibly without cloning. However, since it's not the goal of this PR I'm not pressing on this one

Reply:
Eh, unfortunately we have a lot of stuff like this in our code. It would be nice to merge it without rerunning CI, as it is extra fragile in the last days (due to 500 errors which we receive from some external servers). But if I need to introduce some other changes anyway I will be happy to fix it as well.

Reply:
@matszczygiel @LukasPukenis tried to simplify this function a bit, please check if you like the result.

Reply:
Well, and here are server errors: https://github.com/NordSecurity/libtelio/actions/runs/6773518206/job/18408574733

Reply:
Just would be super nice to do them in a separate PR ;). As it makes reviewers life slightly harder.

Reply:
@Jauler sure, as a rule of course, but as reviewer requested them here, I just added them, the change is still quite small (and split it to separate commits to make reviewer life easier 😁 )